lib/log.bash

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+export ERROR_LOG_LEVEL=1
+export WARNING_LOG_LEVEL=2
+export ALL_LOG_LEVEL=3
+
+function _log_debug()
+{
+  _about 'log a debug message by echoing to the screen. needs LOG_LEVEL >= ALL_LOG_LEVEL'
+  _param '1: message to log'
+  _example '$ _log_debug "Loading plugin git..."'
+  _group 'log'
+
+  [[ "$LOG_LEVEL" -ge $ALL_LOG_LEVEL ]] || return
+  echo -e "${echo_green}$1"
```

Comment:
And one more thing: Do we want to print colors only if the terminal supports color output?

Like documented here? https://unix.stackexchange.com/a/10065/111314

(I know we're not checking for that in other parts of Bash-it, but we might want to start somewhere...)

Reply:
done, you can take a look now